From 090abd2bfad2caf788317f0da11de753d06a276f Mon Sep 17 00:00:00 2001
From: Tilman Blumenbach <tilman+git@ax86.net>
Date: Tue, 29 May 2018 23:25:33 +0200
Subject: [PATCH] Handle removal of cmake macro qt5_use_modules.

That macro was deprecated for some time already and it looks like it has
finally been removed.

See also:

 - Issue: https://github.com/debauchee/barrier/issues/49
 - Patch source: https://gitweb.gentoo.org/repo/gentoo.git/plain/x11-misc/synergy/files/synergy-1.9.1-qt-5.11.patch?id=5dd1f7a3908dec9ae6cf6773acd8ec3b33fc0b2c
---
 src/gui/CMakeLists.txt          | 2 +-
 src/gui/src/ActionDialog.cpp    | 1 +
 src/gui/src/ScreenSetupView.cpp | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 9c902867..12fb5fda 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -22,7 +22,7 @@ add_executable (barrier WIN32
 
 include_directories (./src)
 
-qt5_use_modules (barrier Core Widgets Network)
+target_link_libraries (barrier Qt5::Core Qt5::Widgets Qt5::Network)
 target_compile_definitions (barrier PRIVATE -DBARRIER_VERSION_STAGE="${BARRIER_VERSION_STAGE}")
 target_compile_definitions (barrier PRIVATE -DBARRIER_REVISION="${BARRIER_REVISION}")
 
diff --git a/src/gui/src/ActionDialog.cpp b/src/gui/src/ActionDialog.cpp
index e824ea3c..3565cfbc 100644
--- a/src/gui/src/ActionDialog.cpp
+++ b/src/gui/src/ActionDialog.cpp
@@ -25,6 +25,7 @@
 
 #include <QtCore>
 #include <QtGui>
+#include <QButtonGroup>
 
 ActionDialog::ActionDialog(QWidget* parent, ServerConfig& config, Hotkey& hotkey, Action& action) :
     QDialog(parent, Qt::WindowTitleHint | Qt::WindowSystemMenuHint),
diff --git a/src/gui/src/ScreenSetupView.cpp b/src/gui/src/ScreenSetupView.cpp
index e42ae17e..46e70990 100644
--- a/src/gui/src/ScreenSetupView.cpp
+++ b/src/gui/src/ScreenSetupView.cpp
@@ -22,6 +22,7 @@
 
 #include <QtCore>
 #include <QtGui>
+#include <QHeaderView>
 
 ScreenSetupView::ScreenSetupView(QWidget* parent) :
     QTableView(parent)
-- 
2.17.0

